sklearn/neighbors/lof.py

```diff
@@ -0,0 +1,315 @@
+        Affects only :meth:`kneighbors` and :meth:`kneighbors_graph` methods.
+
+
+    Attributes
+    ----------
+    outlier_factor_ : numpy array, shape (n_samples,)
+        The LOF of X. The lower, the more normal.
+        Inliers tend to have a LOF score close to 1, while outliers tend
+        to have a larger LOF score.
+
+        The local outlier factor (LOF) of a sample captures its
+        supposed 'degree of abnormality'.
+        It is the average of the ratio of the local reachability density of
+        a sample and those of its k-nearest neighbors.
+
+    References
+    ----------
+    .. [1] Breunig, M. M., Kriegel, H. P., Ng, R. T., & Sander, J. (2000, May).
+           LOF: identifying density-based local outliers. In ACM sigmod record.
+    """
+    def __init__(self, n_neighbors=5, algorithm='auto', leaf_size=30,
+                 metric='minkowski', p=2, metric_params=None,
+                 contamination=0.1, n_jobs=1):
+        self._init_params(n_neighbors=n_neighbors,
+                          algorithm=algorithm,
+                          leaf_size=leaf_size, metric=metric, p=p,
+                          metric_params=metric_params, n_jobs=n_jobs)
+
+        self.contamination = contamination
+
+    def fit_predict(self, X, y=None):
+        """Compute the local outlier factor (LOF) on X.
+
+        Return the labels (1 inlier, -1 outlier) of X according to LOF score
+        and the contamination parameter.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features), default=None
+            The query sample or samples to compute the Local Outlier Factor
+            w.r.t. to the training samples.
+
+        Returns
+        -------
+        is_inlier : array, shape (n_samples,)
+            Returns 1 for anomalies/outliers and -1 for inliers.
+        """
+
+        return self.fit(X)._predict()
+
+    def fit(self, X, y=None):
+        """Fit the model using X as training data.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix, BallTree, KDTree}
+            Training data. If array or matrix, shape [n_samples, n_features],
+            or [n_samples, n_samples] if metric='precomputed'.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        if not (0. < self.contamination <= .5):
+            raise ValueError("contamination must be in (0, 0.5]")
+
+        super(LocalOutlierFactor, self).fit(X)
+
+        self._distances_fit_X_, self._neighbors_indices_fit_X_ = (
+            self.kneighbors(None))
+
+        # Compute score over training samples to define threshold_:
+        self.outlier_factor_ = self._local_outlier_factor(
+            self._distances_fit_X_, self._neighbors_indices_fit_X_)
+
+        self.threshold_ = -scoreatpercentile(
+            self.outlier_factor_, 100. * (1. - self.contamination))
+
+        return self
+
+    def _predict(self, X=None):
```

Comment:
I would really like to be consistent. I don't think there's a good argument to have one but not the other. Not sure if the example is a strong enough point to make them both public.